### ISSUE TITLE
no-typeof-undefined should exclude globally declared variables like window

### ISSUE THREAD

**Issue by ian-craig** [palantir/tslint#3642]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.7.0
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
declare var window: any;

if (typeof window !== "undefined") {
    // do something 
}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "no-typeof-undefined": true,
    },
}
```

#### Actual behavior
TSLint gives error

```ts
[tslint] Avoid typeof x === 'undefined' comparisons. Prefer x == undefined or x === undefined: typeof window !== "undefined" (no-typeof-undefined)
```

The suggested fix would be:

```ts
if (window !== undefined) {
    // do something 
}
```

which results in `ReferenceError: 'window' is undefined at...`

#### Expected behavior

Ignore `typeof foo !== "undefiend"` when `foo` has been `declare`d globally.

**Comment by buu700** [comment]
It looks like that rule is actually part of [tslint-microsoft-contrib](https://github.com/Microsoft/tslint-microsoft-contrib), not tslint itself.
